internal/servicelog/formatter.go

```diff
@@ -52,6 +54,57 @@ func NewFormatWriter(dest io.Writer, serviceName string) io.Writer {
 	}
 }
 
+type trimWriter struct {
+	dest     io.Writer
+	re       *regexp.Regexp
+	buf      []byte
+	postTrim bool
+}
+
+func NewTrimWriter(dest io.Writer, regex string) (io.Writer, error) {
+	re, err := regexp.Compile(regex)
+	if err != nil {
+		return nil, err
+	}
+	return &trimWriter{
+		dest: dest,
+		re:   re,
+	}, nil
+}
+
+func (w *trimWriter) Write(p []byte) (nn int, ee error) {
+	// Buffered content has already been searched for newlines - track this.
+	pos := len(w.buf)
+
+	w.buf = append(w.buf, p...)
+	written := 0
+	for {
+		end := bytes.IndexRune(w.buf[pos:], '\n')
+		if end == -1 {
+			return written, nil // wait for rest of line
+		}
+		end += pos
+		loc := w.re.FindIndex(w.buf[:end])
+		start := 0
+		if loc != nil && loc[0] == 0 {
+			start = loc[1]
+		}
```

Comment:
We only want it to match on the left, correct? Can we use a `^` regex anchor, to make it more efficient (so the regex match isn't doing a search, only to test for `loc[0] == 0` after)? We could take the user's `log-trim` value and make the actual regex `"^(" + logTrim + ")"` or similar?

Reply:
For now, I changed it to prepend `^` to the user-supplied regex.